sdcm/scylla_bench_thread.py

```diff
@@ -142,13 +142,77 @@ def __init__(self, stress_cmd, loader_set, timeout, node_list=None, round_robin=
         #       and not really needed having 'coordinated omission fixed' latency one.
         self.hdr_tags = ["co-fixed"]
 
+    def get_datacenter_name_for_loader(self, loader):
+        datacenter_name_per_region = self.loader_set.get_datacenter_name_per_region(db_nodes=self.node_list)
+        datacenter_cmd = ""
+        if loader_dc := datacenter_name_per_region.get(loader.region):
+            datacenter_cmd = f" -datacenter={loader_dc} "
+        else:
+            LOGGER.error("Not found datacenter for loader region '%s'. Datacenter per loader dict: %s",
+                         loader.region, datacenter_name_per_region)
+        return datacenter_cmd
+
+    def adjust_cmd_node_option(self, stress_cmd, loader, cmd_runner):
+        """
+            Adjusts the stress command to include node and rack options based on the current configuration.
+            This method modifies the provided `stress_cmd` string to specify which database nodes and racks
+            should be targeted by the stress tool. It appends node IPs and, if applicable, rack and datacenter
+            information to the command. The adjustments depend on the loader's region, rack, and the test
+            configuration parameters.
+            Args:
+                stress_cmd (str): The initial stress command to be adjusted.
+                loader: The loader node object for which the command is being constructed.
+                cmd_runner: The object used to run commands on the loader.
+            Returns:
+                str: The adjusted stress command string with node and rack options appended as needed.
+        """
+        LOGGER.debug("Start adjusting stress command to use nodes: %s", stress_cmd)
+        dc_options_cmd = ""
+        node_list = self.node_list
+        if self.params.get("rack_aware_loader"):
+            LOGGER.debug("Rack-aware loader is enabled, trying to pin stress command to rack")
+            # if there are multiple rack/AZs configured, we'll try to configue c-s to pin to them
+            rack_names = self.loader_set.get_rack_names_per_datacenter_and_rack_idx(db_nodes=self.node_list)
```

Comment:
On line 171 you assigned the `self.node_list` to the local var `node_list` - it is ok.
But then you continue using the `self.node_list` variant.

Why?

Reply:
Technically speaking, usage of the `node_list` will give you the same result.

Then, considering the motivation for using a separate function, it must be renamed.
For example, to the `node_list_for_current_cmd`.
Having different purpose means different name.

Reply:
changed